src/common/components/publishedField/__tests__/PublishedField.test.jsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import { useRecordContext, useLocaleState } from 'react-admin';
+import get from 'lodash/get';
+
+import PublishedField from '../PublishedField';
+import { toDateTimeString } from '../../../utils';
+
+// Mock the react-admin hooks and lodash/get
+vi.mock('react-admin', () => ({
+  useRecordContext: vi.fn(),
+  useLocaleState: vi.fn(),
+}));
+
+vi.mock('lodash/get', () => ({
+  __esModule: true,
+  default: vi.fn(),
+}));
+
+// Mock utils
+vi.mock('../../../utils', () => ({
+  toDateTimeString: vi.fn(),
+}));
+
+describe('PublishedField', () => {
+  beforeEach(() => {
+    vi.clearAllMocks();
+    useLocaleState.mockReturnValue(['en']); // Default locale
+  });
+
+  it('renders null when no record is present', () => {
+    // Arrange
+    useRecordContext.mockReturnValue(undefined);
+
+    // Act
+    const result = render(<PublishedField source="publishedAt" />);
+
+    // Assert
+    expect(result.container.firstChild).toBeNull();
+  });
+
+  it('renders emptyText when source data is not present', () => {
+    // Arrange
+    const mockRecord = { id: 1 };
+    const emptyText = 'Not published';
+    useRecordContext.mockReturnValue(mockRecord);
+    get.mockReturnValue(undefined);
+
+    // Act
+    render(<PublishedField source="publishedAt" emptyText={emptyText} />);
+
+    // Assert
+    expect(screen.getByText(emptyText)).toBeInTheDocument();
```

Comment:
I feel the `// Arrange`, `// Act` and `// Assert` although bring some clarity, they actually reduce the readability by including information that stops the natural flow of reading the tests. If you want to keep these, it's ok for me, but I'd take another look at them and remove them from where it seems appropriate. But, you do as you see is best in this PR.